packages/plugin-vue/src/handleHotUpdate.ts

```diff
@@ -46,7 +46,14 @@ export async function handleHotUpdate({
     !isEqualBlock(descriptor.script, prevDescriptor.script) ||
     !isEqualBlock(descriptor.scriptSetup, prevDescriptor.scriptSetup)
   ) {
-    affectedModules.add(mainModule)
+    let scriptModule: ModuleNode | undefined
+    if (descriptor.script?.lang && !descriptor.script.src) {
```

Comment:
It's to handle the case where `<script lang='jsx'>` is used, `sfc` request the file `.vue?vue&type=script&lang.jsx`.
When the `jsx` code is changed, the `sfc` file will be requested again.
Because the module of `.vue?vue&type=script&lang.jsx` is excluded by the `handleHotUpdate` of `plugin-vue`,  
`.vue?vue&type=script&lang.jsx` will not be injected with `t= lastHMRTimestamp`, so it will not be reloaded.